added aditional params option

zivOrLive opened the pull request:
I have noticed the lack of ability to send additional environment variables, and so I added an ability to pass a HashMap<String,String> of additional params

jakub-bochenski reviewed (changes requested):
Thanks for the PR, can you please remove the unnecessary formatting changes to make review easier?

zivOrLive commented:
Hey mate, fixed it. Also added a UT. Very appreciated, waiting for your CR

‫בתאריך יום ב׳, 8 בפבר׳ 2021 ב-17:27 מאת ‪jakub-bochenski‬‏ <‪
notifications@github.com‬‏>:‬

> *@jakub-bochenski* requested changes on this pull request.
>
> Thanks for the PR, can you please remove the unnecessary formatting
> changes to make review easier?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jenkinsci/logstash-plugin/pull/98#pullrequestreview-585617886>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKBTYUAWUOXM3EWW3BFR4MLS577F7ANCNFSM4XHKQCHQ>
> .
>

zivOrLive commented:
> Thanks for the PR, can you please remove the unnecessary formatting changes to make review easier?

Yes I fixed it

jakub-bochenski commented:
sorry for the delay, I will try to review this by EOW

jakub-bochenski commented on src/main/java/jenkins/plugins/logstash/persistence/BuildData.java:
```diff
@@ -178,17 +178,23 @@ public int getPassCount()
     rootBuildNum = build.getRootBuild().getNumber();
     buildVariables = build.getBuildVariables();
+    buildVariables = new HashMap<String,String>(buildVariables);
+    buildVariables.putAll(additionalParams);
```
why add it to build variables?

zivOrLive replied:
While, I thought the most natural place of additional parameters, such as environment variables not already included, is in the build variables. Do you disagree or have a better suggestion?

jakub-bochenski replied:
Why not just pass it as a separate field in the output?

zivOrLive replied:
Fair enough

jakub-bochenski commented on src/test/java/jenkins/plugins/logstash/LogstashWriterTest.java:
```diff
@@ -58,19 +60,22 @@ static LogstashWriter createLogstashWriter(final AbstractBuild<?, ?> testBuild,
                                              final BuildData data) {
-    return new LogstashWriter(testBuild, error, null, testBuild.getCharset()) {
+    HashMap<String, String> additionalParams = new HashMap<String, String>();
+    additionalParams.put("param1", "value1");
```
the values are not asserted anywhere, so just use an emptyMap()

jakub-bochenski commented on src/test/java/jenkins/plugins/logstash/PipelineTest.java:
```diff
@@ -128,4 +130,43 @@ public void logstashSend() throws Exception
+    assertThat(data.getString("result"),equalTo("SUCCESS"));
+  }
+
+  public String mapFormatter(Map<String, Object> lhm1,String result) {
```
this doesn't seem to be used?

jakub-bochenski commented on src/main/java/jenkins/plugins/logstash/persistence/BuildData.java:
```diff
@@ -178,17 +178,23 @@ public int getPassCount()
+    this(build, currentTime, listener, null);
+  }
+  // Freestyle project build
+  public BuildData(AbstractBuild<?, ?> build, Date currentTime, TaskListener listener, Map<String, String> additionalParams) {
```
it seems there is no way to use this additional params in freestile builds, right?

zivOrLive replied:
why not?

jakub-bochenski replied:
I don't see any stapler bindings updated, how would it be configured?

zivOrLive replied:
For now it won't be implemented in freestyle builds

jakub-bochenski replied:
OK, can you please remove those extra constructors?
Otherwise LGTM

jakub-bochenski reviewed (changes requested):
Can you please add documentation/example to readme on how to use thix

jakub-bochenski commented on src/test/java/jenkins/plugins/logstash/PipelineTest.java:
```diff
@@ -128,4 +130,43 @@ public void logstashSend() throws Exception
+    assertThat(dataLines.size(), equalTo(1));
+    JSONObject firstLine = dataLines.get(0);
+    JSONObject data = firstLine.getJSONObject("data");
+    assertThat(data.getString("result"),equalTo("SUCCESS"));
```
I see no assertion checking that the parameters are included

jakub-bochenski commented on src/main/java/jenkins/plugins/logstash/persistence/BuildData.java:
```diff
@@ -469,4 +491,4 @@ public String getAgentName() {
     this.agentName = agentName;
   }
-}
+}
```
minor: no newline at end

jakub-bochenski commented on src/main/java/jenkins/plugins/logstash/pipeline/LogstashSendStep.java:
```diff
@@ -110,4 +121,4 @@ public String getFunctionName()
     }
   }
-}
+}
```
minor: no newline at end

jakub-bochenski commented:
@zivOrLive do you intend to finish work on this. 

Do you need  help?

zivOrLive commented:
> @zivOrLive do you intend to finish work on this.
> 
> Do you need help?

Im not sure how to adjust it so you could use it in freestyle builds, if you could direct me it would be of much help (Im pretty new to jenkins)

jakub-bochenski commented:
It's OK, it can be a Pipeline only feature at first. 
I just needs some short update to the documentation with example use in pipeline (and a notice that it doesn't work in Freestyle)

jakub-bochenski commented on src/main/java/jenkins/plugins/logstash/persistence/BuildData.java:
```diff
@@ -176,19 +176,26 @@ public int getPassCount()
     rootBuildNum = build.getRootBuild().getNumber();
     buildVariables = build.getBuildVariables();
+    // buildVariables = new HashMap<String,String>(buildVariables);
+    // buildVariables.putAll(additionalParams);
```
please remove the commented out code

jakub-bochenski reviewed (changes requested):
Please apply those 2 more corrections, otherwise LGTM

jakub-bochenski commented:
@mwinter69 please take a look if you have time/feel like it :)
